Bio/SeqIO/Interfaces.py

```diff
@@ -58,21 +58,17 @@ def __init__(self, source, alphabet=None, mode="t", fmt=None):
             else:
                 raise ValueError(f"Unknown mode '{mode}'") from None
             self.stream = source
-            self.should_close_stream = False
         try:
             self.records = self.parse(self.stream)
         except Exception:
-            if self.should_close_stream:
-                self.stream.close()
+            self.stream.close()
```

Comment:
You do not need to close the stream here.

Reply:
I updated the as per your suggestion.